engine/classes/Elgg/HooksRegistrationService.php

```diff
@@ -90,6 +90,33 @@ public function unregisterHandler($name, $type, $callback) {
 	}
 
 	/**
+	 * Removes all handlers
+	 *
+	 * @param string   $name
+	 * @param string   $type
+	 *
+	 * @return array
+	 * @access private
+	 */
+	public function clearHandlers($name, $type = null) {
+		$handlers = array();
+		
+		if ($type !== null) {
+			if (isset($this->handlers[$name][$type])) {
+				$handlers = $this->handlers[$name][$type];
+				unset($this->handlers[$name][$type]);
+			}
+		} else {
+			if (isset($this->handlers[$name])) {
+				$handlers = $this->handlers[$name];
+				unset($this->handlers[$name]);
+			}
+		}
+		
+		return $handlers;
```

Comment:
I feel like we should return these in the same structure as the internal storage. E.g. clear('foo', 'bar') might return `['foo' => ['bar' => handlers]]`. That could allow easier reinjection should we add that later.